course_info/static/course_info/js/controllers/IndexController.js

```diff
@@ -124,9 +142,55 @@
 
         $scope.initializeDatatable = function() {
             $scope.dataTable = $('#courseInfoDT').DataTable({
-                data: $scope.courseInfo,
-                dataSrc: '',  // data is an array
-                // todo: p_r_ocessing?...
+                serverSide: true,
+                deferLoading: true,
+                ajax: function(data, callback, settings) {
+                    $scope.$apply(function(){
+                        $scope.searchInProgress = true;
+                    });
+                    var queryParameters = {};
+                    if ($scope.queryString.trim() != '') {
+                        queryParameters.search = $scope.queryString.trim();
+                    }
+                    if ($scope.filtersApplied) {
+                        for (var key in $scope.filters) {
+                            var f = $scope.filters[key];
+                            if (f.query) { queryParameters[f.key] = f.value; }
+                        }
+                    }
+                    queryParameters.offset = data.start;
+                    queryParameters.limit = data.length;
+                    var order = data.order[0];
+                    queryParameters.ordering = (order.dir == 'desc' ? '-' : '') + $scope.columnFieldMap[order.column];
+                    $.ajax({
+                        url: '/icommons_rest_api/api/course/v2/course_instances',
+                        method: 'GET',
+                        data: queryParameters,
+                        dataType: 'json',
+                        success: function(data, textStatus, jqXHR) {
+                            var results = data.results;
+                            var resultsLength = data.results.length;
+                            var processedData = [];
+                            for (var i = 0; i < resultsLength; i++) {
+                                processedData.push($scope.courseInstanceToTable(results[i]));
+                            }
+                            callback({
+                                recordsTotal: data.count,
+                                recordsFiltered: data.count,
```

Comment:
I think our API returns `count` as the number of records found after applying the query parameters. DataTables has an ability to show records filtered (I don't think we're enabling it in this UI), but it's ambiguous whether when using serverside processing that returned `count` should be recordsFiltered or recordsTotal (which might be, say, the total count of all CourseInstances unfiltered by the query parameters). I think recordsFiltered makes more sense when doing client-side filtering (because it should be effectively instantaneous), but really it's the same thing in the end, the filtering is just being applied in different places.